src/core/balance-sheet.ts

```diff
@@ -1,29 +1,55 @@
 import { getBalance as getBitcoindBalance } from "@services/bitcoind"
 import { lndsBalances } from "@services/lnd/utils"
 import { baseLogger } from "@services/logger"
-import { User } from "@services/mongoose/schema"
+import { InvoiceUser, User } from "@services/mongoose/schema"
 import { ledger } from "@services/mongodb"
 
 import { WalletFactory } from "./wallet-factory"
 
 const logger = baseLogger.child({ module: "balanceSheet" })
 
+export const updatePendingLightningTransactions = async () => {
+  let user, userWallet
+
+  // select distinct user ids from pending invoices
+  const usersWithPendingInvoices = InvoiceUser.aggregate([
+    { $match: { paid: false } },
+    { $group: { _id: "$uid" } },
+  ])
+    .cursor({ batchSize: 100 })
+    .exec()
```

Comment:
this query is just distinct

Reply:
I'm not sure to understand what you mean here?

Reply:
This is like `select distinct uid from invoiceusers where paid = false`. This query only get the users that have pending invoices

Reply:
I had to [look this up](https://codeburst.io/javascript-array-distinct-5edc93501dc4) as well ... I think it means there are no duplicate values in the array(?) that's returned

Reply:
yup ok thanks for clarifying